docs/mp/persistence.adoc

```diff
@@ -0,0 +1,1430 @@
+
+[source,yaml]
+----
+javax:
+  sql:
+    DataSource:
+      yourDataSourceName:
+        somePropertyOfYourConnectionPoolAndDataSource: itsValue
+        someOtherPropertyOfYourConnectionPoolAndDataSource: anotherValue
+----
+
+===== Example: Configuring the Oracle Universal Connection Pool and Oracle JDBC [[DS-Configuration-UCP-Oracle-Example]]
+
+This example presumes you have:
+
+* <<DS-UCP-Project-Setup,set up the Oracle Universal Connection Pool>>
+
+* <<DS-Oracle-Driver-Project-Setup,set up Oracle JDBC>>
+
+This example, in Java properties file format, configures an Oracle
+Universal Connection Pool-managed data source named `main` to
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-C4F2CA86-0F68-400C-95DA-30171C9FB8F0[connect
+to an Oracle Database] on `localhost` port `1521`, using the
+`oracle.jdbc.poolOracleDataSource` vendor-supplied `DataSource`, with
+a service name of `XE`, a `user` of `scott`, and a `password` of
+`tiger`:
+
+[source,properties]
+----
+# (Why "connectionFactoryClassName"? See
+# https://docs.oracle.com/en/database/oracle/oracle-database/21/jjuar/oracle/ucp/jdbc/PoolDataSourceImpl.html#setConnectionFactoryClassName_java_lang_String_)
+javax.sql.DataSource.main.connectionFactoryClassName = oracle.jdbc.pool.OracleDataSource
+javax.sql.DataSource.main.url = jdbc:oracle:thin://@localhost:1521/XE # see https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-EF07727C-50AB-4DCE-8EDC-57F0927FF61A
+javax.sql.DataSource.main.user = scott
+javax.sql.DataSource.main.password = tiger
+----
+
+In general, the properties that can be set on the Oracle Universal
+Connection Pool can be inferred from the "setter" methods found in
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jjuar/oracle/ucp/jdbc/PoolDataSourceImpl.html[the
+javadoc for the `PoolDataSourceImpl` class].
+
+In general, the properties that can be set on the
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/jdbc/pool/OracleDataSource.html[`oracle.jdbc.pool.OracleDataSource`]
+`DataSource` implementation can be inferred from the "setter" methods
+found in
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/jdbc/pool/OracleDataSource.html[its
+javadoc].
+
+NOTE: https://github.com/brettwooldridge/HikariCP/blob/HikariCP-5.0.1/src/main/java/com/zaxxer/hikari/util/PropertyElf.java#L46-L53[Unlike
+HikariCP], the Oracle Universal Connection Pool does not distinguish
+cleanly between configuration properties that affect _its_ behavior
+and those that affect the behavior of the vendor-supplied `DataSource`
+implementation whose connections it pools.  For example, in the
+example above it is not possible to tell that
+https://docs.oracle.com/en/database/oracle/oracle-database/21/jjuar/oracle/ucp/jdbc/PoolDataSourceImpl.html#setConnectionFactoryClassName_java_lang_String_[`connectionFactoryClassName`
```

Comment:
Perhaps you could a base-url attribute for `https://docs.oracle.com/en/database/oracle/oracle-database/21` ; maybe that belong in includes/attributes.adoc